# seq-al
## Aligning DNA and Protein sequences using Dynamic Programming
 
The following algorithms are implemented
 1. Needleman-Wunsch Algorithm for global sequence alignment with linear gap penalties.
 2. Modified Gotoh's Algorithm for global sequence alignment with affine gap penalties.
 3. Smith-Waterman Algorithm for local sequence alignment with linear gap penalties.
 
*References*:
1. [Rolf Backofen (2010). Lecture Slides for Bioinformatics I — SS 2010. *Institut für Informatik.*](http://helios.mi.parisdescartes.fr/~lomn/Cours/BI/Material2019/gap-penalty-gotoh.pdf)
2. [Flouri, T., Kobert, K., Rognes, T., & Stamatakis, A. (2015). Are all global alignment algorithms and implementations correct? *BioRxiv.*](https://doi.org/10.1101/031500)
3. [Altschul, S. F., & Erickson, B. W. (1986). Optimal sequence alignment using affine gap costs. *Bulletin of Mathematical Biology.*](https://doi.org/10.1007/BF02462326)
